src/main/java/de/tum/in/www1/artemis/domain/hestia/ProgrammingExerciseTask.java

```diff
@@ -0,0 +1,96 @@
+package de.tum.in.www1.artemis.domain.hestia;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.DomainObject;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.ProgrammingExerciseTestCase;
+
+/**
+ * A ProgrammingExerciseTask
+ */
+@Entity
+@Table(name = "programming_exercise_task")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class ProgrammingExerciseTask extends DomainObject {
+
+    @Column(name = "task_name")
+    private String taskName;
+
+    // No orphanRemoval here, as there should only be one parent-child relationship (which is ProgrammingExercise -> CodeHint)
```

Comment:
We don't really need the hint for the exercise relationship. If the task gets deleted the hint should usually also be deleted.
We had this in orphanRemoval before and Stephan told us to change this with this exact reasoning. The reason behind this is apparently to prevent issues when deleting an entire exercise, as this could lead to Hibernate trying to delete the same hint twice.

Reply:
I see, breaking the loop.